pkg/registry/core/namespace/storage/storage.go

```diff
@@ -147,6 +147,16 @@ func (r *REST) Delete(ctx context.Context, name string, options *metav1.DeleteOp
 		)
 		return nil, false, err
 	}
+	if options.Preconditions.ResourceVersion == nil {
+		options.Preconditions.ResourceVersion = &namespace.ResourceVersion
```

Comment:
same comment here... if the API request did not specify a resourceversion, we should not pin a precondition to the pre-fetched instance's resourceversion here